### PR TITLE
Add publish pipeline for dotnet emitter

### PR DESCRIPTION
**Pull request by m-nash**
Fixes https://github.com/microsoft/typespec/issues/3111

### CODE REVIEW

File: packages/http-client-csharp/eng/pipeline/publish.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+trigger: none
+pr: none
+
+extends:
+  template: /eng/pipelines/templates/1es-redirect.yml
+  parameters:
+    variables:
+      BlobFeedUrl: https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json
```
Comment (weshaggard) [inline packages/http-client-csharp/eng/pipeline/publish.yml:8]: Is this used?

Comment (m-nash) [reply packages/http-client-csharp/eng/pipeline/publish.yml:8]: no, I was just checking in a file to test the pipeline triggers

**Comment by m-nash** [comment]
Blocked by this issue https://github.com/microsoft/typespec/issues/3117

File: eng/emitters/pipelines/templates/archetype-preview.yml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      # Responsible for building the autorest generator and typespec emitter packages
+      # Produces the artifact `build_artifacts` which contains the following:
+      #   emitter-package.json: Will be used to generate the emitter-package-lock.json file and placed in the `/eng` folder
+      #                         of the sdk repository.
+      #   overrides.json: Contains npm package version overrides for the emitter and generator
+      #   packages/: Contains the packages to publish
+      - stage: Build
+        variables:
+          selfRepositoryPath: $(Build.SourcesDirectory)
+        jobs:
```
Comment (hallipr) [inline eng/emitters/pipelines/templates/archetype-preview.yml:71]:
```suggestion
      - stage: Build
        variables:
          selfRepositoryPath: $(Build.SourcesDirectory)
        pool:
          name: $(LINUXPOOL)
          image: $(LINUXVMIMAGE)
          os: linux
        jobs:
```

like here:
https://github.com/microsoft/typespec/blob/main/eng/pipelines/ci.yml#L121C1-L124C20

**Comment by m-nash** [comment]
Follow up item https://github.com/microsoft/typespec/issues/3132

File: eng/tsp-core/pipelines/ci.yml
Comment (timotheeguerin) [inline eng/tsp-core/pipelines/ci.yml:1]: rename this `publish.yml`

File: eng/emitters/pipelines/variables/image.yml
Comment (timotheeguerin) [inline eng/emitters/pipelines/variables/image.yml:1]: was this file meant to be moved under emitters? shouldn't it be shared if you also need it?

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**